### DISCUSSION TITLE
Organization search is not working properly

### DISCUSSION THREAD

**Discussion by somvitbhowmik** [Sunbird-Lern/Community#83] [Query]
Hi Team,
Organization search is not working properly. For example:
Consider that we have to search **Block Development Officer** organization.

    if I search the organization with **Block** keyword. We are getting 1 result.
    Next we are trying to search **Block Development** keyword. We are getting multiple resuts. Total results count is 7.
    Now if we search with the complete sentence **Block Development Officer**.

It will fetch the data based on the word **Block** then **Development** and **Officer** and return all the results, Which is fine. But in a response it is very difficult to find the **Block Development Officer** organization. Because getting huge response. Because we are searching for the Block Development Officer and it should get the high preference and should show 1st in the response. Which is not happening. Organization search command sharing for the reference.

`curl --location --request POST 'https://<Host:Port>/api/org/v1/search' \ --header 'Content-Type: application/json' \ --header 'Authorization: <Bearer Token>' \ --data-raw '{ "request": { "filters": { "isTenant": true, "status": 1 }, "sortBy": { "orgName": "asc" }, "query": "Block Development Officer", "limit": 500 } }'`

@reshmi-nair Kindly let me know any solution for this problem.

cc: @pkranga

Regards,
Somvit

**Comment by reshmi-nair** [comment]
@somvitbhowmik Need to check the feasibility of this in Elastic Search. But we are currently working on few planned items. So you are welcome to contribute to the SB Lern - UserOrg service if you have any solution.
